src/pages/cart/Cart.js

```diff
@@ -1,7 +1,158 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useMediaQuery } from 'react-responsive';
+import toast from 'react-hot-toast';
+import { Typography, Button, BookCard } from '../../components';
+import ItemCounter from '../../components/common/ItemCounter/ItemCounter';
+import { calculateTotalPrice, setItem } from '../../utils/helperFuncs';
+import { getCart, removeFromCart, updateCart } from '../../services/flixycartApi';
+import './Cart.scss';
 
 const Cart = () => {
-  return <div>Cart</div>;
+  const [cart, setCart] = useState([]);
+  const sm = useMediaQuery({ maxWidth: '600px' });
+
+  useEffect(() => {
+    const getCartDetails = async () => {
+      const res = await getCart();
+      if (res.status === 200) {
+        setCart(res.data.result);
+      }
+    };
+    getCartDetails();
+  }, []);
+
+  const onCartCountChange = useCallback(async ({ type, id, quantity }) => {
+    const res = await updateCart({ type, id, quantity });
+    if (res.status === 200) {
+      setCart(res.data.result);
+    }
+  }, []);
+
+  const handleRemoveFromCart = async (e, id) => {
+    e.preventDefault();
+    const res = await removeFromCart(id);
+    if (res.status === 200) {
+      setCart(res.data.result);
+    }
+    e.stopPropagation();
+  };
+
+  const moveToWishlist = (e, item) => {
+    e.preventDefault();
+    const wishlistData = [];
+    wishlistData.push(item);
```

Comment:
I was reading tanays chart of pitfalls to avoid and it mentioned there that use impure ways to update array and object so that react to render the changes it mentioned there that instead of push use concat or spread or state